### PR TITLE
SAGE-677 Changed to Ubuntu 20 OS, K3S isntalled

### PR DESCRIPTION
**Pull request by ozorob2**

### CODE REVIEW

File: ROOTFS/media/rpi/sage-utils/dhcp-pxe/tftp/cmdline.txt [hunk 1]
```diff
@@ -1 +1 @@
-console=serial0,115200 console=tty1 root=/dev/nfs nfsroot=10.31.81.1:/media/rpi/sage-utils/dhcp-pxe/nfs,vers=3 ro ip=dhcp rootwait elevator=deadline init=/bin/ro-root.sh
+console=serial0,115200 console=tty1 root=/dev/nfs nfsroot=10.31.81.1:/media/rpi/sage-utils/dhcp-pxe/nfs,vers=3 ro ip=dhcp rootwait elevator=deadline init=/bin/ro-root.sh cgroup_memory=1 cgroup_enable=memory
```
Comment (jswantek) [inline ROOTFS/media/rpi/sage-utils/dhcp-pxe/tftp/cmdline.txt:1]: Do you still need these cgroup changes for Ubuntu?  

Are you sure this `cmdline.txt` file is still being used?  I sorta assumed the boot procedure for ubuntu would be very different then Raspbian ?  maybe not....

Also, the `init=/bin/ro-root.sh' part?  i assumed this was Raspbian only and didn't apply to Ubuntu.  This is what got me thinking if this cmdline.txt file is even being used anymore.

You can check on your running system by looking at `/proc/cmdline` or something

File: version [hunk 2]
```diff
@@ -1 +1 @@
-1.1.5
+1.1.6
```
Comment (jswantek) [inline version:1]: This is a non-backwards compatible change and therefore the version should be bumped to 2.0.0

File: release.sh [hunk 3]
```diff
@@ -69,7 +69,12 @@ mkdir -p ${BASEDIR}/media/rpi/sage-utils/dhcp-pxe/nfs
 
 cp -pr bootmnt/* ${BASEDIR}/media/rpi/sage-utils/dhcp-pxe/tftp/
 cp -pr rootmnt/* ${BASEDIR}/media/rpi/sage-utils/dhcp-pxe/nfs/
-cp -pr ROOTFS/media/rpi/sage-utils/dhcp-pxe/* ${BASEDIR}/media/rpi/sage-utils/dhcp-pxe/
+cp -pr ROOTFS/media/rpi/sage-utils/dhcp-pxe/* ${BASEDIR}/media/rpi/sage-utils/dhcp-pxe
+zcat ${BASEDIR}/media/rpi/sage-utils/dhcp-pxe/tftp/vmlinuz > ${BASEDIR}/media/rpi/sage-utils/dhcp-pxe/tftp/vmlinux
+
+wget https://github.com/rancher/k3s/releases/download/v1.20.0+k3s2/k3s-arm64
```
Comment (jswantek) [inline release.sh:75]: In addition to getting the binary can you also curl the script itself (from https://get.k3s.io/) and put it in `${BASEDIR}/media/rpi/sage-utils/dhcp-pxe/nfs/etc/installK3S.sh` instead of committing a static version of the file?

I can see a reason to commit a static version of the k3s installer shell script.... getting it dynamically has the advantage of not needing to commit it to our code repo.  But we would not get any bug fixes dynamically.  We would also not get any new bugs....

Basically, if you feel that since u are getting a specific version of k3s (v1.20.0) that you also want a static version of the installer script then I would be okay with keeping it checked in.

File: ROOTFS/media/rpi/sage-utils/dhcp-pxe/nfs/etc/installK3S.sh [hunk 4]
```diff
@@ -0,0 +1,864 @@
+#!/bin/sh
```
Comment (jswantek) [inline ROOTFS/media/rpi/sage-utils/dhcp-pxe/nfs/etc/installK3S.sh:1]: See my comment in `release.sh` about downloading this file at build-time instead of committing a version of the file into our code repository.

File: ROOTFS/media/rpi/sage-utils/dhcp-pxe/nfs/etc/rc.local [hunk 5]
```diff
@@ -25,17 +25,19 @@ FDISK_CMDS
...
-
 #Installing all packages from /isodebs/
 dpkg -i /isodebs/*
 echo "Packages In /isodebs/ Installed" | xargs -L 1 echo `date +'[%Y-%m-%d %H:%M:%S]'` >> /etc/rc.local.logs
+
+mkdir -p /media/plugin-data/k3s/kubelet
+mkdir -p /media/plugin-data/k3s/rancher
+ln -s /media/plugin-data/k3s/kubelet/ /var/lib/kubelet
+ln -s /media/plugin-data/k3s/rancher/ /var/lib/rancher
+INSTALL_K3S_SKIP_DOWNLOAD=true /etc/installK3S.sh
```
Comment (jswantek) [inline ROOTFS/media/rpi/sage-utils/dhcp-pxe/nfs/etc/rc.local:43]: Please add a log to `/etc/rc.local.logs` at the end of this script indicating that things were successful

File: ROOTFS/media/rpi/sage-utils/dhcp-pxe/tftp/cmdline.txt (hunk 1 above)
Comment (ozorob2) [reply ROOTFS/media/rpi/sage-utils/dhcp-pxe/tftp/cmdline.txt:1]: The cgroup's were still a necessary change, was getting error messages from k3s about them

File: version (hunk 2 above)
Comment (ozorob2) [reply version:1]: Will do

File: release.sh (hunk 3 above)
Comment (ozorob2) [reply release.sh:75]: Yeah, I think it's best to keep this version of the installer script checked in since we don't know how future installer scripts may look, they could rename some parameters that would make this not work

File: ROOTFS/media/rpi/sage-utils/dhcp-pxe/nfs/etc/rc.local (hunk 5 above)
Comment (ozorob2) [reply ROOTFS/media/rpi/sage-utils/dhcp-pxe/nfs/etc/rc.local:43]: will do
